app/kubemci/cmd/create.go

```diff
@@ -145,9 +153,18 @@ func runCreate(options *CreateOptions, args []string) error {
 
 	// Unmarshal the YAML into ingress struct.
 	var ing v1beta1.Ingress
-	if err := unmarshallAndApplyDefaults(options.IngressFilename, &ing); err != nil {
+	if err := unmarshallAndApplyDefaults(options.IngressFilename, options.namespace, &ing); err != nil {
 		return fmt.Errorf("error in unmarshalling the yaml file %s, err: %s", options.IngressFilename, err)
 	}
+	if options.StaticIPName != "" {
+		addAnnotation(&ing, annotations.StaticIPNameKey, options.StaticIPName)
+	}
+	ingAnnotations := annotations.IngAnnotations(ing.Annotations)
+	// We require the specification of a static IP as an annotation.
+	if ingAnnotations == nil || ingAnnotations.StaticIPName() == "" {
+		return fmt.Errorf("Ingress spec must provide a Global Static IP through annotation of %q (alternatively, use --static-ip flag)", annotations.StaticIPNameKey)
```

Comment:
We start the error with lowercase, since the calling method can prefix it to add more context.

Reply:
done